cms/models/pluginmodel.py

```diff
@@ -357,6 +383,31 @@ def get_children(self):
     def get_descendants(self):
         return CMSPlugin.objects.filter(pk__in=self._get_descendants_ids())
 
+    def get_ancestors(self):
+        """
+        Retrieve the list of ancestor plugins for the current plugin.
+
+        This method returns a list of ancestor plugins, starting from the root
+        ancestor down to the immediate parent of the current plugin. If the
+        current plugin has no parent, an empty list is returned.
+
+        Returns:
+            list: A list of ancestor plugins, ordered from the root ancestor
+                  to the immediate parent of the current plugin.
+        """
+        if not self.parent_id:
+            return []
+        if self._state.fields_cache.get('parent') or not plugin_supports_cte():
+            return self.parent.get_ancestors() + [self.parent]
+        return list(self._get_ancestors_from_db())
```

Comment:
I really would like to get a `QuerySet` instead of a list. Otherwise I have to filter in code rather than in the database.